app/models/listing.rb

```diff
@@ -1,2 +1,4 @@
 class Listing < ApplicationRecord
+  has_many_attached :photos
+  validates :location, :description, :title, :price, presence: true
```

Comment:
@solowresidual for now I think you can remove the `location` field since I replaced it with additional address fields...